### ISSUE TITLE
Problems in version 2.5.3（issue）

### ISSUE THREAD

**Issue by QAQlinmiao** [haubna/PhysicsMod#81]
Other creatures have monsters holding objects or tools and cannot display the objects they hold after they die.
other problems：
I think the doll's movements are too stiff
When i kick it
His limbs can't bend naturally, they don't look natural.
Besides,
Sometimes the rag doll will fall into the bedrock layer

**Comment by haubna** [comment]
Can you record cases where you see them falling through blocks onto the bedrock layer? I don't know a proper solution for the stiffness yet because it only appears stiff due to the arms getting blocked by the head and thus they can't move properly. Holding items are not a priority right now.

**Comment by QAQlinmiao** [comment]
https://user-images.githubusercontent.com/83525912/130322359-c703ba03-89e3-4083-b785-39ea6f581c27.mp4

**Comment by QAQlinmiao** [comment]
https://user-images.githubusercontent.com/83525912/130319699-59683b41-0530-47aa-9da4-5479073f0873.mp4


https://user-images.githubusercontent.com/83525912/130319711-20c1e6d9-4e61-4c04-b547-f5ba7e35c965.mp4

You should be able to see it now，Actually the knees and hands should be bent, but they still cannot be bent，It should be unable to bend, so it becomes the picture below
![2021-08-19_22 33 41](https://user-images.githubusercontent.com/83525912/130308003-4ad5a85b-c346-4497-ba1d-a4c10559b648.png)

It feels a bit like being fixed by screws
So he can't bend naturally
The hand will also be raised instead of lying on the ground naturally
The feet will not open naturally but are glued together

**Comment by QAQlinmiao** [comment]
Oh right
One more thing I didn't mention
Is the doll in the water
In the game the magma block should cause the ragdoll to be sucked to the bottom of the sea
Soul sand will cause the doll to be washed into the sea
But it’s not reflected in the game
I Hope to fix it

**Comment by haubna** [comment]
> Oh right
> Can physics be applied to the head
> One more thing I didn't mention
> Is the doll in the water
> In the game the magma block should cause the ragdoll to be sucked to the bottom of the sea
> Soul sand will cause the doll to be washed into the sea
> But it’s not reflected in the game
> I Hope to fix it QWQ

Oh so you are pushing the ragdolls into the void. Not much I can do there. I will have a look at the stiffness though.
Should be an easy fix to let the physics react to soul sand and magma block water. Thanks for the reports :)

**Comment by QAQlinmiao** [comment]
emmm, in fact, I didn't want to push them into the void, but I just don't know why it would jam and get squeezed into it.
The squares from the ground to the bedrock layer feel like they can’t be blocked, it feels like there is empty (like air), I just don’t want them to fall

**Comment by QAQlinmiao** [comment]
The skeleton holding the enchanted bow has no bow in his hand after death

**Comment by haubna** [comment]
> emmm, in fact, I didn't want to push them into the void, but I just don't know why it would jam and get squeezed into it.
> The squares from the ground to the bedrock layer feel like they can’t be blocked, it feels like there is empty (like air), I just don’t want them to fall

This is due to performance optimizations. It only adds blocks to the physics that aren't obscured by solid blocks.

**Comment by QAQlinmiao** [comment]
> > emmm，其实我不想把它们推入虚空，只是不知道为什么会卡住，挤进去。
> > 从地面到基岩层的方块感觉都挡不住，感觉像是空的（像空气），我只是不想让它们掉下来
> 
> 这是由于性能优化。它只会向未被实体块遮挡的物理添加块。

Oh that's it

**Comment by QAQlinmiao** [comment]
Close this question now
Test version 2.5.4 first
There is a problem reporting to you

**Comment by QAQlinmiao** [comment]
It seems that the problem of stiffness has gotten some relief 
Of course you can also refer to this MOD Ragdoll Corpses （1.8.1）
